homeassistant/components/climate/zha.py

```diff
@@ -0,0 +1,211 @@
+                          update_before_add=True)
+    else:
+        _LOGGER.warning("ZigBee Climate %s %s not supported yet",
+                        discovery_info['manufacturer'],
+                        discovery_info['model'])
+
+
+class ClimateST218(zha.Entity, ClimateDevice):
+    """Climage Stelpro ST218"""
+    _domain = DOMAIN
+
+    def __init__(self, **kwargs):
+        """Initialize the ZHA Climate."""
+        super().__init__(**kwargs)
+
+        self._modes = {0: 'off',
+                       4: 'confort',
+                       5: 'eco',
+                       }
+        self._reverse_modes = dict((v, k) for k, v in self._modes.items())
+
+        self._target_temperature = None
+        self._current_temperature = None
+        self._current_operation = None
+        self._heating_power = None
+        self._operation_list = list(self._modes.values())
+        self._unit = 'C'
+
+        # Add Stelpro Specific
+        import bellows.types as t
+        # pylint: disable=W0212
+        self._endpoint.thermostat._attridx['setpoint_mode_manuf_specific'] = \
+            0x401C
+        # pylint: enable=W0212
+        self._endpoint.thermostat.attributes[0x401C] = \
+            ('setpoint_mode_manuf_specific', t.int8s)
+        self._endpoint.thermostat.attributes[0x001C] = \
+            ('system_mode', t.uint8_t)
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve latest state."""
+
+        @asyncio.coroutine
+        def safe_read(cluster, attributes, raw=False):
+            """Swallow all exceptions from network read.
+            If we throw during initialization, setup fails. Rather have an
+            entity that exists, but is in a maybe wrong state, than no entity.
+            """
+            try:
+                result, _ = yield from cluster.read_attributes(
+                    attributes,
+                    allow_cache=False,
+                    raw=raw,
+                )
+                return result
+            except Exception:  # pylint: disable=broad-except
+                return {}
+
+        _LOGGER.debug("%s async_update", self.entity_id)
+
+        result = yield from safe_read(self._endpoint.thermostat,
+                                      ['local_temp',
+                                       'pi_heating_demand',
+                                       'occupied_heating_setpoint',
+                                       'setpoint_mode_manuf_specific'])
+        self._heating_power = int(result.get('pi_heating_demand'))
+        raw_mode = result.get('setpoint_mode_manuf_specific', {})
+        self._current_operation = self._modes.get(raw_mode)
+
+        raw_temp = result.get('occupied_heating_setpoint')
+        if raw_temp:
+            self._target_temperature = raw_temp / 100.0
+        raw_temp = result.get('local_temp')
+        if raw_temp:
+            self._current_temperature = raw_temp / 100.0
+
+    @property
+    def temperature_unit(self):
```

Comment:
As far as I can see, `self._unit` is set exactly once, on initialization, and always to "C". So this should always return TEMP_CELSIUS? Is `self._unit` still necessary then?

Reply:
Fixed